app/app/src/main/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenter.java

```diff
@@ -34,17 +35,50 @@ public void onClearButtonPress() {
         view.handleOperations(false);
     }
 
+    public void onDeleteButtonPress() {
+        String reduction = EMPTY_STRING;
+
+        if (model.getOperator().isEmpty()) {
+            reduction = model.getFirstNumber();
+            if (!reduction.isEmpty()) {
+                reduction = reduction.substring(0, reduction.length() - 1);
+                model.setFirstNumber(reduction);
+                view.showNumberPressed(model.getFirstNumber());
+                controlDot(model.getFirstNumber());
+            }
+        } else {
+            reduction = model.getSecondNumber();
+            if (!reduction.isEmpty()) {
+                reduction = reduction.substring(0, reduction.length() - 1);
+                model.setSecondNumber(reduction);
+                view.showNumberPressed(model.getSecondNumber());
+                controlDot(model.getSecondNumber());
```

Comment:
this code is repeated at line 43, you can create a function using model.get#Number as parameter

Reply:
done, thanks